django/db/models/fields/related.py

```diff
@@ -1866,8 +1880,8 @@ class ForeignKey(ForeignObject):
     }
     description = _("Foreign Key (type determined by related field)")
 
-    def __init__(self, to, to_field=None, related_name=None, related_query_name=None,
-            limit_choices_to=None, parent_link=False, on_delete=CASCADE,
+    def __init__(self, to, on_delete, related_name=None, related_query_name=None,
```

Comment:
I don't think it's possible to hit the second deprecation warning. `on_delete` needs to remain a kwarg so it's optional, but after that, something like `models.ForeignKey(Question, 'pub_date', on_delete=models.PROTECT)` will raise `TypeError: __init__() got multiple values for keyword argument 'on_delete'`.